bindings/rust/s2n-tls-tokio/tests/common/mod.rs

```diff
@@ -95,3 +95,25 @@ where
     );
     Ok((client?, server?))
 }
+
+pub async fn get_tls_streams<A: Builder, B: Builder>(
+    server_builder: A,
+    client_builder: B,
+) -> Result<
+    (
+        TlsStream<TcpStream, A::Output>,
+        TlsStream<TcpStream, B::Output>,
+    ),
+    Box<dyn std::error::Error>,
+>
+where
+    <A as Builder>::Output: Unpin,
+    <B as Builder>::Output: Unpin,
+{
+    let (server_stream, client_stream) = get_streams().await?;
+    let connector = TlsConnector::new(client_builder);
+    let acceptor = TlsAcceptor::new(server_builder);
+    let (client_tls, server_tls) =
+        run_negotiate(&connector, client_stream, &acceptor, server_stream).await?;
+    Ok((server_tls, client_tls))
+}
```

Comment:
nit: I don't think we ever use this functions with connection pools, so using concrete types would simplify this quite a bit

```suggestion
pub async fn get_tls_streams(
    server_builder: config::Config,
    client_builder: config::Config,
) -> Result<(TlsStream<TcpStream>, TlsStream<TcpStream>), Box<dyn std::error::Error>> {
    let (server_stream, client_stream) = get_streams().await?;
    let connector = TlsConnector::new(client_builder);
    let acceptor = TlsAcceptor::new(server_builder);
    let (client_tls, server_tls) =
        run_negotiate(&connector, client_stream, &acceptor, server_stream).await?;
    Ok((server_tls, client_tls))
}
```

Reply:
I really don't think simplifying a common test library method is worth potentially needing to modify it later if we do want to use a different kind of Builder. People really shouldn't need to care about the exact definition of get_tls_streams, it'll just work. It also matches the existing [run_negotiate](https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls-tokio/tests/common/mod.rs#L80-L97) common test library method.